pkg/resolver/search.go

```diff
@@ -509,41 +293,46 @@ func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
 			whereDs = append(whereDs, goqu.L(`"value"`).ILike(key).Expression())
 		}
 	}
+
 	if input.Filters != nil {
 		for _, filter := range input.Filters {
 			if len(filter.Values) > 0 {
 				values := pointerToStringArray(filter.Values)
 
-				// If property is of array type like label, remove the equal sign in it and use colon
-				// - to be similar to how it is stored in the database
-				if _, ok := arrayProperties[filter.Property]; ok {
-					cleanedVal := make([]string, len(values))
-					for i, val := range values {
-						labels := strings.Split(val, "=")
-						if len(labels) == 2 {
-							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
-						} else if len(labels) == 1 {
-							//// If property is of array type, format it as an array for easy searching
-							cleanedVal[i] = labels[0]
-						} else {
-							klog.Error("Error while decoding label string")
-							cleanedVal[i] = val
+				if len(propTypeMap) > 0 {
+					if dataTypeInMap, ok := propTypeMap[filter.Property]; !ok { // check if value exists
+						klog.Warningf("Property type for [%s] doesn't exist in cache. Refreshing property type cache",
+							filter.Property)
+						propTypeMapNew, err := getPropertyType(ctx, true) // Refresh the property type cache.
+						if err != nil {
+							klog.Warningf("Error creating property type map with err: [%s] ", err)
 						}
+						propTypeMap = propTypeMapNew
+						break
 
-					}
-					values = cleanedVal
-				}
+					} else {
+						klog.V(5).Infof("Prop in map:%s, filter prop is: %s, datatype :%s\n", dataTypeInMap, filter.Property)
```

Comment:
```suggestion
						klog.V(5).Infof("For filter prop: %s, datatype is :%s\n", filter.Property, dataTypeInMap)
```
Remove the extra %s.

Reply:
that's more clear! ty!